### DISCUSSION TITLE
Neural Filters

### DISCUSSION THREAD

**Discussion by Shaarigan** [LinSoftWin/Photoshop-CC2022-Linux#104] [Q&A] [answered]
Hey,

is there any way to get Neural Filters working on Linux?

**Comment by MiMillieuh** [comment]
If it's related to GPU acceleration it won't likely work... if not try disabling GPU acceleration

**Reply by Shaarigan** [reply]
Neural filters are disabled for me in the menu

**Reply by MiMillieuh** [reply]
I've tried on both Nvidia and AMD GPU and they are grayed out. I guess that feature require some specific GPU accelerations stuff that aren't implemented.

GPU acceleration implementations are really tricky as in a video game you use API like DirectX or Vulkan, Compute acceleration that Photoshop use is tied to the Driver (Nvidia or AMD or Intel) that means that unless this has been reverse engineered, it won't likely work in any circumstances.

The thing to remember is that Compute acceleration isn't equals to GPU acceleration even thought Photoshop calls them both GPU acceleration.

**Reply by Shaarigan** [reply]
In the meantime I found out that it is requiered for Neural Filters to be logged in into the CC account or else it will not allow to use them. Even if you're logged in, chances are high that it isn't able to verify properly but will show as if this is a test version of PS and Neural Filters will work only once per session then.

I did not try to do this on Linux yet because I have doubts that it will have a negative impact on my Adobe subscription. Do you have any idea about that?

**Reply by MiMillieuh** [reply, marked as answer]
Well to answer that it's not possible to login with CC in Linux... 

CC login uses a IE implementation that doesn't work at all on Linux (that's the same for the installer) So it's literally impossible to login for now.

And if it was possible I don't think adobe will do something with your account, plugins are usually allowed so I don't think running it on Linux will kill your subscription. (But still I don't know. I can't use my subscription too since it's not possible to login... I can with lightroom but the web version.)

**Reply by Shaarigan** [reply]
That's sad, I'd liked to try their AI stuff but well 🤷

**Reply by MiMillieuh** [reply]
> That's sad, I'd liked to try their AI stuff but well shrug

Well AI stuff is already unstable when used as intended so I guess it would be wors... but yeah I don't know how it would work with wine tbh...

If you really need it you could use a VM with a lightweight windows (VMware has kinda good graphic performances or if you're really determinated you could do a GPU passthrough with KVM)

Sometimes when I need AE I use it in a VM and it do the trick as long as the project isn't too big

**Reply by Shaarigan** [reply]
I don't need it per se but it makes some tasks easier and some tutorials already only relyon these filters
